### DISCUSSION TITLE
>  NX   Running target prep for 78 projects failed Tasks not run because their dependencies failed or --nx-bail=true:

### DISCUSSION THREAD

**Discussion by shiv9604** [storybookjs/storybook#25315] [Help] [answered]
### Summary

Hi Team, I had set up the storybook repo by forking it. While yarn starts I am getting the following error and my application is not served, Anyone before faced this error, Can someone please help me to complete my setup by fixing the error.

**Error :** 

NX   Running target prep for 78 projects failed

   Tasks not run because their dependencies failed or --nx-bail=true:

   - @storybook/web-components-webpack5:prep
   - @storybook/web-components-vite:prep
   - @storybook/server-webpack5:prep
   - @storybook/builder-manager:prep
   - @storybook/web-components:prep
   - @storybook/preset-server-webpack:prep
   - @storybook/vue3-vite:prep
   - @storybook/html-vite:prep
   - @storybook/sveltekit:prep
   - @storybook/addon-interactions:prep
   - @storybook/addon-backgrounds:prep
   - @storybook/angular:prep
   - @storybook/addon-storysource:prep
   - @storybook/addon-essentials:prep
   - @storybook/nextjs:prep
   - @storybook/addon-viewport:prep
   - @storybook/addon-controls:prep
   - @storybook/addon-toolbars:prep
   - @storybook/manager-api:prep
   - @storybook/core-server:prep
   - @storybook/addon-actions:prep
   - @storybook/addon-measure:prep
   - @storybook/addon-outline:prep
   - @storybook/components:prep
   - @storybook/addon-themes:prep
   - @storybook/addon-links:prep
   - @storybook/addon-docs:prep
   - @storybook/addon-jest:prep
   - @storybook/addon-a11y:prep
   - @storybook/manager:prep
   - @storybook/blocks:prep
   - @storybook/cli:prep

   Failed tasks:

   - @storybook/theming:prep

   View structured, searchable error logs at https://nx.app/runs/atJrfoTcrP

    at makeError (c:\Users\shivk\Desktop\Projects\Open Source\storybook\scripts\node_modules\execa\lib\error.js:59:11)
    at handlePromise (c:\Users\shivk\Desktop\Projects\Open Source\storybook\scripts\node_modules\execa\index.js:119:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at exec (c:\Users\shivk\Desktop\Projects\Open Source\storybook\scripts\utils\exec.ts:41:7)
    at runTask (c:\Users\shivk\Desktop\Projects\Open Source\storybook\scripts\task.ts:318:24)
    at run (c:\Users\shivk\Desktop\Projects\Open Source\storybook\scripts\task.ts:464:28) {
  shortMessage: 'Command failed with exit code 1: nx run-many --target="prep" --all --parallel --max-parallel=7 -- --reset',
  command: 'nx run-many --target="prep" --all --parallel --max-parallel=7 -- --reset',
  escapedCommand: '"nx run-many --target=\\"prep\\" --all --parallel --max-parallel=7 -- --reset"',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: '\r\n' +
    ' >  NX   Running target prep for 78 projects:\r\n' +
    '\r\n' +
    '    - @storybook/web-components-webpack5\r\n' +
    '    - @storybook/preset-web-components-webpack\r\n' +
    '    - @storybook/web-components-vite\r\n' +
    '    - @storybook/preact-webpack5\r\n' +
    '    - @storybook/svelte-webpack5\r\n' +
    '    - @storybook/server-webpack5\r\n' +
    '    - @storybook/react-webpack5\r\n' +
    '    - @storybook/builder-webpack5\r\n' +
    '    - @storybook/preset-create-react-app\r\n' +
    '    - @storybook/vue3-webpack5\r\n' +
    '    - @storybook/builder-manager\r\n' +
    '    - @storybook/web-components\r\n' +
    '    - @storybook/html-webpack5\r\n' +
    '    - @storybook/preset-server-webpack\r\n' +
    '    - @storybook/preset-preact-webpack\r\n' +
    '    - @storybook/svelte-vite\r\n' +
    '    - @storybook/preact-vite\r\n' +
    '    - @storybook/preset-svelte-webpack\r\n' +
    '    - @storybook/builder-vite\r\n' +
    '    - @storybook/react-vite\r\n' +
    '    - @storybook/preset-react-webpack\r\n' +
    '    - @storybook/preset-html-webpack\r\n' +
    '    - @storybook/vue3-vite\r\n' +
    '    - @storybook/html-vite\r\n' +
    '    - @storybook/sveltekit\r\n' +
    '    - @storybook/preset-vue3-webpack\r\n' +
    '    - @storybook/addon-interactions\r\n' +
    '    - @storybook/addon-backgrounds\r\n' +
    '    - @storybook/angular\r\n' +
    '    - @storybook/react-dom-shim\r\n' +
    '    - @storybook/addon-storysource\r\n' +
    '    - @storybook/source-loader\r\n' +
    '    - @storybook/addon-essentials\r\n' +
    '    - @storybook/client-logger\r\n' +
    '    - @storybook/nextjs\r\n' +
    '    - @storybook/svelte\r\n' +
    '    - @storybook/instrumenter\r\n' +
    '    - @storybook/server\r\n' +
    '    - @storybook/core-webpack\r\n' +
    '    - @storybook/ember\r\n' +
    '    - @storybook/preact\r\n' +
    '    - @storybook/addon-highlight\r\n' +
    '    - @storybook/preview-api\r\n' +
    '    - @storybook/addon-viewport\r\n' +
    '    - @storybook/core-events\r\n' +
    '    - @storybook/addon-controls\r\n' +
    '    - @storybook/core-common\r\n' +
    '    - @storybook/addon-toolbars\r\n' +
    '    - @storybook/manager-api\r\n' +
    '    - @storybook/core-server\r\n' +
    '    - @storybook/node-logger\r\n' +
    '    - @storybook/react\r\n' +
    '    - @storybook/vue3\r\n' +
    '    - @storybook/addon-actions\r\n' +
    '    - @storybook/docs-tools\r\n' +
    '    - @storybook/html\r\n' +
    '    - @storybook/addon-measure\r\n' +
    '    - @storybook/addon-outline\r\n' +
    '    - @storybook/csf-plugin\r\n' +
    '    - @storybook/telemetry\r\n' +
    '    - @storybook/components\r\n' +
    '    - @storybook/csf-tools\r\n' +
    '    - @storybook/addon-themes\r\n' +
    '    - @storybook/addon-links\r\n' +
    '    - @storybook/channels\r\n' +
    '    - @storybook/addon-docs\r\n' +
    '    - @storybook/addon-jest\r\n' +
    '    - @storybook/addon-a11y\r\n' +
    '    - @storybook/codemod\r\n' +
    '    - @storybook/theming\r\n' +
    '    - @storybook/preview\r\n' +
    '    - @storybook/addon-mdx-gfm\r\n' +
    '    - @storybook/manager\r\n' +
    '    - @storybook/router\r\n' +
    '    - @storybook/blocks\r\n' +
    '    - @storybook/types\r\n' +
    '    - @storybook/test\r\n' +
    '    - @storybook/cli\r\n' +
    '   \r\n' +
    '   With additional flags:\r\n' +
    '     --reset=true\r\n' +
    '\r\n' +
    ' \r\n' +
    '\r\n' +
    '> nx run @storybook/client-logger:prep --reset\r\n' +
    '\r\n' +
    '(node:12340) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/node-logger:prep --reset\r\n' +
    '\r\n' +
    '(node:15660) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/core-events:prep --reset\r\n' +
    '\r\n' +
    '(node:18012) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/addon-mdx-gfm:prep --reset\r\n' +
    '\r\n' +
    '(node:17824) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/channels:prep --reset\r\n' +
    '\r\n' +
    '(node:23088) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/router:prep --reset\r\n' +
    '\r\n' +
    '(node:10528) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/theming:prep --reset\r\n' +
    '\r\n' +
    '(node:23224) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    '\n' +
    '> node -r C:\\Users\\shivk\\Desktop\\Projects\\Open Source\\storybook\\scripts\\prepare/../node_modules/esbuild-register/register.js ./scripts/fix-theme-type-export.ts\n' +
    'node:internal/modules/cjs/loader:1080\r\n' +
    '  throw err;\r\n' +
    '  ^\r\n' +
    '\r\n' +
    "Error: Cannot find module 'C:\\Users\\shivk\\Desktop\\Projects\\Open'\r\n" +
    'Require stack:\r\n' +
    '- internal/preload\r\n' +
    '\x1B[90m    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)\x1B[39m\r\n' +
    '\x1B[90m    at Module._load (node:internal/modules/cjs/loader:922:27)\x1B[39m\r\n' +
    '\x1B[90m    at internalRequire (node:internal/modules/cjs/loader:174:19)\x1B[39m\r\n' +
    '\x1B[90m    at Module._preloadModules (node:internal/modules/cjs/loader:1433:5)\x1B[39m\r\n' +
    '\x1B[90m    at loadPreloadModules (node:internal/process/pre_execution:611:5)\x1B[39m\r\n' +
    '\x1B[90m    at setupUserModules (node:internal/process/pre_execution:130:3)\x1B[39m\r\n' +
    '\x1B[90m    at prepareExecution (node:internal/process/pre_execution:113:5)\x1B[39m\r\n' +
    '\x1B[90m    at prepareMainThreadExecution (node:internal/process/pre_execution:40:3)\x1B[39m\r\n' +
    '\x1B[90m    at node:internal/main/run_main_module:10:1\x1B[39m {\r\n' +
    "  code: \x1B[32m'MODULE_NOT_FOUND'\x1B[39m,\r\n" +
    "  requireStack: [ \x1B[32m'internal/preload'\x1B[39m ]\r\n" +
    '}\r\n' +
    '\r\n' +
    'Node.js v18.18.2\r\n' +
    '\x1B[31mAn error occurred while executing: `node -r C:\\Users\\shivk\\Desktop\\Projects\\Open Source\\storybook\\scripts\\prepare/../node_modules/esbuild-register/register.js ./scripts/fix-theme-type-export.ts`\x1B[39m\n' +
    'undefined\n' +
    '\n' +
    'Error: Command failed with exit code 1: node -r C:\\Users\\shivk\\Desktop\\Projects\\Open Source\\storybook\\scripts\\prepare/../node_modules/esbuild-register/register.js ./scripts/fix-theme-type-export.ts\n' +
    '    at makeError (c:\\Users\\shivk\\Desktop\\Projects\\Open Source\\storybook\\scripts\\node_modules\\execa\\lib\\error.js:59:11)\n' +
    '    at handlePromise (c:\\Users\\shivk\\Desktop\\Projects\\Open Source\\storybook\\scripts\\node_modules\\execa\\index.js:119:26)\n' +
    '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n' +
    '    at exec (c:\\Users\\shivk\\Desktop\\Projects\\Open Source\\storybook\\scripts\\utils\\exec.ts:41:7)\n' +
    '    at run (c:\\Users\\shivk\\Desktop\\Projects\\Open Source\\storybook\\scripts\\prepare\\bundle.ts:152:5)\n' +
    '\r\n' +
    '> nx run @storybook/types:prep --reset\r\n' +
    '\r\n' +
    '(node:11416) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/csf-tools:prep --reset\r\n' +
    '\r\n' +
    '(node:2544) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/react-dom-shim:prep --reset\r\n' +
    '\r\n' +
    '(node:21960) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/core-common:prep --reset\r\n' +
    '\r\n' +
    '(node:4364) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/preset-create-react-app:prep --reset\r\n' +
    '\r\n' +
    '(node:17148) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/source-loader:prep --reset\r\n' +
    '\r\n' +
    '(node:14996) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/csf-plugin:prep --reset\r\n' +
    '\r\n' +
    '(node:21896) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/codemod:prep --reset\r\n' +
    '\r\n' +
    '(node:22684) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/core-webpack:prep --reset\r\n' +
    '\r\n' +
    '(node:17920) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/preview-api:prep --reset\r\n' +
    '\r\n' +
    '(node:14136) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/telemetry:prep --reset\r\n' +
    '\r\n' +
    '(node:18984) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/preset-preact-webpack:prep --reset\r\n' +
    '\r\n' +
    '(node:21796) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/preset-web-components-webpack:prep --reset\r\n' +
    '\r\n' +
    '(node:12448) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/preset-svelte-webpack:prep --reset\r\n' +
    '\r\n' +
    '(node:23100) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/preset-html-webpack:prep --reset\r\n' +
    '\r\n' +
    '(node:21640) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings ...` to show where the warning was created)\n' +
    'done\n' +
    '\r\n' +
    '> nx run @storybook/docs-tools:prep --reset\r\n' +
    '\r\n' +
    '(node:2104) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time\n' +
    '(Use `node --trace-warnings'... 6873 more characters,
  stderr: '',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}

**Comment by Marklb** [comment, marked as answer]
I don't know if this is a Windows problem, but it looks like you are using Windows and it is recommended to use WSL for Windows. Most of the project should work, but some things may not, such as the sandboxes. I haven't tried the project outside of WSL much recently, but last time I tried, the tests worked and sandboxes did not work.

As for the error, in one of the error messages, I see `Error: Cannot find module 'C:\Users\shivk\Desktop\Projects\Open'`, which seems to be cutting the path off early. If that is the case, then it probably is a bug, but may work if you don't use a path that has space.

I tried to reproduce the problem, on Windows, and it was reproducible by adding a space to one of the files in my path. I didn't dig into if this is a simple fix or not, though.

So, as a workaround, if you put your project in a folder that will not have any parent folders containing a space then it should work.
